### PR TITLE
feat: Install operators in batch.

### LINKED ISSUE #624: Deploy operator CR in batch
A follow up issue for https://github.com/IBM/operand-deployment-lifecycle-manager/issues/622#issuecomment-790136949

@horis233

### PR DESCRIPTION
**Pull request by horis233**
user can use the flag "-batch-chunk-size" to control how many subscriptions will be deployed simultaneously

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #624 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -73,6 +73,8 @@ func main() {
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	var stepSize = flag.Int("batch-chunk-size", 3, "batch-chunk-size is used to control at most how many subscriptions will be created concurrently")
```
Comment (Daniel-Fan) [inline main.go:76]: May I ask that if `batch-chunk-size` is the flag, does it mean we need pre-set it in our CSV as well?
https://github.com/IBM/operand-deployment-lifecycle-manager/blob/master/bundle/manifests/operand-deployment-lifecycle-manager.clusterserviceversion.yaml#L283

Comment (horis233) [reply main.go:76]: It has a default value "3". It is not a necessary flag

Comment (Daniel-Fan) [reply main.go:76]: Oh I see, thanks!

**Comment by horis233** [comment]
/hold this PR 

1_bZHBo75FSyKre5pk2-HPmw

Too many locks used in this change, we need a better method to manage the batch install.

**Comment by horis233** [comment]
@Daniel-Fan @ZhuoxiLi 

This PR is ready for review.

This PR is changed much from the last review. It still uses locks to update status. I tried pushing status update into a channel, but the effect is not very good.

Let's use the lock as version 1. We can make further refactor in future PRs.

**Comment by horis233** [comment]
Looks, it didn't pass the e2e test. I will make more testings on it

**Comment by horis233** [comment]
@Daniel-Fan @ZhuoxiLi 

It finally passes all the tests. Please review this PR. 

We can take it the stage 1 of the concurrence support. We can do more reactors on it in the future.

**Review by Daniel-Fan — APPROVED** [review]
/lgtm

**Pull request merged**